airflow/configuration.py

```diff
@@ -69,7 +69,9 @@ def run_command(command):
     process = subprocess.Popen(
         shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True
     )
-    output, stderr = [stream.decode(sys.getdefaultencoding(), 'ignore') for stream in process.communicate()]
+    output, stderr = [
+        stream.decode(sys.getdefaultencoding(), 'ignore') for stream in process.communicate(timeout=60)
```

Comment:
I wonder if this one (and maybe some others) need to be configurable.

Reply:
This value is much greater than the time it takes to complete this process. It seems to me that processes that take longer than 5 seconds will not be started here as this would have an impact on the environment performance.  I am not trying to solve the problems that processes are freezing and we should handle these situations somehow, but for now I just want the user to be notified that the process has stopped responding. In the future, we might think of a better way to handle it once these problems are reported along with the use case.

What do you think about increasing this time to 5 minutes? Is this a more rational time for you?

Reply:
This case is to run command from the `*_CMD` vars, for instance `sqlalchemy_conn_cmd` setting.

60s is fine for most sensible cases (probably too long), but we just can't know what users might put in these configs -- it might take a long time on first instance.

Perhaps we just document this as a limitation for now and we can make it configurable when someone complains.

Reply:
I added more docs now :-)